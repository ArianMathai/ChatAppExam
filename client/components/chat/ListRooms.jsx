import {useEffect, useState} from "react";
import {Link} from "react-router-dom";
import NewChat from "./NewChat";

function ListRooms(){

    const [rooms, setRooms] = useState([]);

    async function fetchRooms(){

        const res = await fetch("/api/chat/getAllRooms");

        if (res.ok){
            const data = await res.json();
            console.log("Rooms data = ", data)
            setRooms(data)
        }
        else{
            console.log("Failed to fetch rooms")
        }

    }

    useEffect(() => {
        fetchRooms();
    }, []);



    return (
        <>
            <ul>
                {rooms.map((r, index) => (
                    <li key={index}>
                        <Link to={`/chatroom/${r.roomName}`}>
                            <div>{r.roomName}</div>
                        </Link>
                    </li>
                ))}
            </ul>
            <div>
                <Link to={"/chat/new"}>
                    <button>New chat</button>
                </Link>
            </div>
        </>
    );


}
export default ListRooms;