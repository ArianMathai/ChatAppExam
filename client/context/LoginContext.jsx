import React from "react";

export const LoginContext = React.createContext({
    user: undefined,
    username: undefined,
    google_client_id: undefined,
    google_discovery_url: undefined,
    microsoft_endpoint: undefined,
    microsoft_client_id: undefined,
    loadUser: async () => {},
});