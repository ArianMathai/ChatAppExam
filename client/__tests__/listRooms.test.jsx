import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ListRooms from '../components/chat/ListRooms';
import { LoginContext } from '../context/LoginContext';


global.fetch = jest.fn(() =>
    Promise.resolve({
        ok: true,
        json: () =>
            Promise.resolve([
                { roomName: 'MockedRoom1', participants: ['user1', 'user2'] },
                { roomName: 'MockedRoom2', participants: ['user1', 'user3'] },
            ]),
    })
);


jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

describe('ListRooms Component', () => {
    it('renders rooms correctly', async () => {
        let component;
        const username = 'brukernavn';

        await act(async () => {
            component = renderer.create(
                <LoginContext.Provider value={{ username: username }}>
                    <ListRooms />
                </LoginContext.Provider>
            );
        });

        expect(component).toMatchSnapshot();
    });
});
