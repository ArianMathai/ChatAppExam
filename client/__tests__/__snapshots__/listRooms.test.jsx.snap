// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRooms Component renders rooms correctly 1`] = `
[
  <h1
    className="chatroom-header"
  >
    Chatrooms
  </h1>,
  <ul
    className="list-room-ul"
  />,
  <div
    className="button-container"
  >
    <a
      href="/chat/new"
    >
      <button>
        New chat
      </button>
    </a>
  </div>,
]
`;
